function-app/adb-to-purview/src/Function.Domain/Helpers/OlProcessing/ValidateOlEvent.cs

```diff
@@ -29,18 +29,39 @@ public ValidateOlEvent(ILoggerFactory loggerFactory)
             _log = loggerFactory.CreateLogger<ValidateOlEvent>();
         }
 
+        /// <summary>
+        /// Helper function to determine if the event is one of
+        /// the data source v2 ones which need to aggregate data
+        /// from the start and complete events
+        /// </summary>
+        private bool isDataSourceV2Event(Event olEvent) {
+            string[] special_cases = {"azurecosmos://", "iceberg://"}; // todo: make this configurable?
+            // Don't need to process START events here as they have both inputs and outputs
+            if (olEvent.EventType == "START") return false;
+
+            foreach (var outp in olEvent.Outputs)
+            {
+                foreach (var source in special_cases)
+                {
+                    if (outp.NameSpace.StartsWith(source)) return true;
+                }
+            }
+            return false;
+        }
+
         /// <summary>
         /// Performs initial validation of OpenLineage input
         /// The tested criteria include:
         /// 1. Events have both inputs and outputs
+        ///     a. Except for special cases covered in isDataSourceV2Event
```

Comment:
Could we add why this is necessary? Such as "OpenLineage does not emit a start event or complete event with all of the necessary information and so must be consolidated later on"?